alexandria/core/models.py

```diff
@@ -240,19 +240,32 @@ def make_checksum(bytes_: bytes) -> str:
         return f"sha256:{hashlib.sha256(bytes_).hexdigest()}"
 
     def set_checksum(self):
+        if not settings.ALEXANDRIA_ENABLE_CHECKSUM:
+            return
+
         self.content.file.file.seek(0)
         self.checksum = self.make_checksum(self.content.file.file.read())
 
     def set_content_vector(self):
+        if not (
+            self.variant == File.Variant.ORIGINAL
+            and settings.ALEXANDRIA_ENABLE_CONTENT_SEARCH
+        ):
```

Comment:
[De Morgan's rule](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) might make it more readable/understandable

```suggestion
        if (
            self.variant != File.Variant.ORIGINAL
            or not settings.ALEXANDRIA_ENABLE_CONTENT_SEARCH
        ):
```